metals/src/main/scala/scala/meta/internal/metals/codeactions/BracelessBracefulSwitchCodeAction.scala

```diff
@@ -0,0 +1,558 @@
+                termTry.expr,
+                "try expression"
+              ),
+              createCodeActionForCatchP(
+                termTry,
+                path,
+                document,
+                "catch expression"
+              ),
+              termTry.finallyp.flatMap(finallyp =>
+                createCodeActionForPotentialBlockHolder(
+                  finallyp,
+                  path,
+                  document,
+                  finallyp,
+                  "finally expression"
+                )
+              )
+            ).flatten
+          case caseTree: Case =>
+            createCodeActionForCasesListWrapper(caseTree, path).toSeq
+          case termIf: Term.If =>
+            Seq(
+              createCodeActionForPotentialBlockHolder(
+                termIf.thenp,
+                path,
+                document,
+                termIf.thenp,
+                "then expression"
+              ),
+              createCodeActionForPotentialBlockHolder(
+                termIf.elsep,
+                path,
+                document,
+                termIf.elsep,
+                "else expression"
+              )
+            ).flatten
+          case termFor: Term.For =>
+            createCodeActionForPotentialBlockHolder(
+              termFor,
+              path,
+              document,
+              termFor.body,
+              "for expression"
+            ).toSeq
+          case termForYield: Term.ForYield =>
+            createCodeActionForPotentialBlockHolder(
+              termForYield,
+              path,
+              document,
+              termForYield.body,
+              "yield expression"
+            ).toSeq
+          case _: Term.Match => None.toSeq
+          case termWhile: Term.While =>
+            createCodeActionForPotentialBlockHolder(
+              termWhile,
+              path,
+              document,
+              termWhile.body,
+              "while expression"
+            ).toSeq
+          case _: Defn.GivenAlias => None.toSeq
+          //          case template: Template =>
+          //            createCodeActionForTemplateHolder(
+          //              template,
+          //              path,
+          //              document,
+          //              template,
+          //              "template"
+          //            ).toSeq
+          case termBlock: Term.Block =>
+            createCodeActionForPotentialBlockHolder(
+              termBlock,
+              path,
+              document,
+              termBlock,
+              "block"
+            ).toSeq
+        }
+      }
+    result.toSeq.flatten
+  }
+
+  private def hasBraces(template: Template): Boolean = {
+    util
+      .Try(template.stats.maxBy(_.pos.end).pos.end)
+      .getOrElse(-1) != template.pos.end && util
+      .Try(template.tokens.maxBy(_.pos.end).text)
+      .getOrElse("") == "}"
+  }
+
+  private def hasBraces(termBlock: Term.Block): Boolean = {
+    util
```

Comment:
`text(termBlock.pos.start)` should always point at '{'

Reply:
aha, No this would create another bug in Scala 3. Because the open brace might be for one of the stats rather Than the whole block. So always it should be checked that the open brace or the closing brace is not part of one of the stats.

Reply:
Some as above then. We should also check if the `{` doesn't belong to the first stat